Version/Version.cpp

```diff
@@ -0,0 +1,96 @@
+#include <filesystem>
+#include <Windows.h>
+#include <iostream>
+
+std::filesystem::path get_exe_path()
+{
+    TCHAR buffer[MAX_PATH];
+    GetModuleFileNameW(NULL, buffer, _countof(buffer));
+    return std::filesystem::path(buffer).parent_path();
+}
+
+std::filesystem::path find_api_directory()
+{
+    return get_exe_path()/(TEXT(R"(ArkApi)"));
+}
+
+std::filesystem::path find_dll()
+{
+    return find_api_directory()/(TEXT(R"(AsaApi.dll)"));
+}
+
+std::filesystem::path get_sys_path()
+{
+	TCHAR sys_dir[MAX_PATH];
+	GetSystemDirectoryW(sys_dir, MAX_PATH);
```

Comment:
TCHAR is a macro for char / wchar_t depending on the UNICODE define.

You can change TCHAR to just wchar_t since we're directly using GetSystemDirectoryW which is always the Unicode version.